src/services/services.ts

```diff
@@ -3345,6 +3347,20 @@ export function createLanguageService(
         );
     }
 
+    function getImports(fileName: string): readonly string[] {
+        synchronizeHostData();
+        const file = getValidSourceFile(fileName);
+        const imports: Set<string> = new Set();
+        let specifiers = file.imports;
+        specifiers = specifiers.slice(specifiers.findIndex(nodeIsSynthesized) + 1);
+        if (specifiers.length === 0) return [];

```

Comment:
return emptyArray

Reply:
I thought emptyArray was slower than `[]`. Also, what happens if somebody mutates the array returned from getImports? Is that a concern for services?